File: PL_Project/perl/program2.pl
#!/bin/perl

#initial balance
$checking = 1000.00;
$saving = 1000.00;

$welcome="*** Welcome to Cal Poly' ATM ***";

print $welcome, "\n";

$count=1;
# Checking for pin here
while (1) {
  # Get user input
  print "Please enter your PIN: ";
  $input=<STDIN>;

  #if it equals the pin then exit
  if($input == 111){  last; }
  system clear;
  if( $count == 3){
    print "Too many illegal PINs. Try later again.\n";
    exit 1;
  }
  $count++;
  
}

system clear;

# while loop for menu
$stop=0;
while($stop == 0){
  menu();
  print "==> Please select option (1-6): ";
  $input = <STDIN>;
  use v5.10.1;
  given ($input) {
    when (1) { checkToSave(); }
    when (2) { saveToCheck();}
    when (3) { 
      print "Saving's account balance: \$ $saving \n"; 
      sleep 1;}
    when (4) { 
      print "Checking's account balanace: \$ $checking \n";
      sleep 1; }
    when (5) { withdraw();
      sleep 1;}
    when (6) { $stop=1; }
    default       { }
  }
  system clear;
}

print "Thank you for using the ATM System.\n";

# Main Menu
sub menu(){
  print "$welcome\n";
  print "(1) Transfer from checking account to savings account\n";
  print "(2) Transfer from savings account to checking account\n";
  print "(3) Savings account balance\n";
  print "(4) Checkings account balance\n";
  print "(5) Withdraw Cash from either account\n";
  print "(6) Exit\n";
}

# Transfer from checkings to savings
sub checkToSave(){
  $amount = getInput("transfer");
  if($amount == 0){
    return;
  }
  # check if amount is greater than balance or neg 
  elsif( $amount > $checking || $amount < 0) {
    print "Transcation not completed\n";
    sleep 1;
  }else{
    print "Transcation completed\n";
    $checking = $checking - $amount;
    $saving = $saving + $amount;
    sleep 1;
  }
}

# Transfer from savings to checkings
sub saveToCheck(){
  $amount = getInput("transfer");
  if($amount == 0){
    return;
  }
  # check if amount is greater than balance or neg 
  elsif( $amount > $saving || $amount < 0) {
    print "Transcation not completed\n";
    sleep 1;
  }else{
    print "Transcation completed\n";
    $checking = $checking + $amount;
    $saving = $saving - $amount;
  }
}

#Withdraw from either account
sub withdraw(){

  print "\nAccounts:\n";
  print "1. Savings Account\n";
  print "2. Checking Account\n";
  print "[Enter any other number to return to main menu]\n\n";
  print "Which account would you like to withraw from? ";
  $account = <STDIN>;

  if( $account == 1){
    $amount = getInput("withdraw");

    # check if amount is greater than balance or neg 
    if($amount > $saving || $amount < 0){
      print "Transaction not completed\n";
    } else{
      $saving = $saving - $amount;
      print "\nTransaction completed\n";
      print "\$$amount has been withdrawn\n";
      print "You have \$$saving remaining in this account\n";
      sleep 2;
    }
  } elsif($account == 2){
    $amount = getInput("withdraw");
    if($amount > $checking || $amount < 0){
      print "Transaction not completed\n";
    } else{
      $checking = $checking - $amount;
      print "\nTransaction completed\n";
      print "\$$amount has been withdrawn\n";
      print "You have \$$saving remaining in this account\n";
      sleep 2;
    }
  } else{
    return;
  }
  sleep 1;
}

# Get input from user
sub getInput(){
  print "\nHow much money would you like to @_? \n";
  print "[To return to the main menu enter zero]\n";
  print "Amount \$ ";
  $amount = <STDIN>;
  return $amount;
}

File: PL_Project/cpp/program2.c
#include <iostream>
#include <iomanip>
using namespace std;

bool pinPrompt(void);
void menu(void);
void checkToSave(float*, float*);
void saveToCheck(float*, float*);
void withdraw(float*, float*);
void withdrawMoney(float*);

int main() {
		int option;
		// Bool will be equal to whether user knows pin
		bool proceed = pinPrompt();
		float savings, checking;
		savings = checking = 1000;

		if (proceed) {
			do {

				// Swicth which is determine by user's selection
				std::system ( "clear" );
				menu();
				cin >> option;

				switch (option) {
					case 1:
						checkToSave(&checking, &savings);
						break;
					case 2:
						saveToCheck(&checking, &savings);
						break;
					case 3:
						cout << "\nYour savings's account balance is $" << setprecision(2) << fixed << savings << "\n";
						sleep(1);
						break;
					case 4:
						cout << "\nYour checking's account balance is $" << setprecision(2) << fixed << checking << "\n";
						sleep(1);
						break;
					case 5:
						withdraw(&checking, &savings);
						break;
					case 6:
						std::system ( "clear" );
						cout << "Thank you for using the ATM system.\n";
						break;
					default:
						cout << "\nInvalid Option. Please choose between 1- 6\n";
   		 			sleep(1);
   		 			break;
					}
			} while (option != 6);
		} else {
				cout << "Too many illegal PINs. Try again later." << endl;
		}
		return 0;
}

// prompt for pin
bool pinPrompt() {
		int pin, attempt = 0;
		cout << "*** Welcome to Cal Poly's ATM ***\n";
		while (attempt < 3) {
				cout << "Please enter your PIN: ";
				cin >> pin;
				if (pin == 111) return true;
				attempt++;
				std::system ( "clear" );
		}
		return false;
}

// menu for the program
void menu() {

				cout << "***   Welcome to Cal Poly's ATM System***\n";
				cout << "(1) Transfer from checking account to savings account\n";
				cout << "(2) Transfer from savings account to checking account\n";     
				cout << "(3) Savings account balance\n";
				cout << "(4) Checking account balance\n";     
				cout << "(5) Withdraw cash from either account\n";
				cout << "(6) Exit\n";
				cout << "==> Please select an option (1-6) : ";
						
		return;
}

// Transfering money from check to saving
void checkToSave(float* checkPtr, float* savePtr) {
		float amount;

		cout << "\nHow much would you like to transfer from checking to savings?\n";
		cout << "[To return to the main menu enter zero]\n";
		cout << "Amount $ ";
		cin >> amount;
		

		if (amount == 0) {
				return;
		}
		// checks for amount entered by use to not be neg and less than balance
		else if (amount > *checkPtr || amount < 0) {
				cout << "\nTransaction not completed.";
				sleep(1);
				return;
		} else {
				*checkPtr -= amount;
				*savePtr += amount;
				cout << "\nTransaction completed\n";
				sleep(1);
		}
		return;
}
// Transfering money from saving to check
void saveToCheck(float* checkPtr, float* savePtr) {
		float amount;

		cout << "\nHow much would you like to transfer from savings to checking?\n";
		cout << "[To return to the main menu enter zero]\n";
		cout << "Amount $ ";
		cin >> amount;
		if (amount == 0) {
				return;
		}
		// checks for amount entered by use to not be neg and less than balance
		if (amount > *savePtr || amount < 0) {
				cout << "\nTransaction not completed.\n";
				sleep(1);
				return;
		} else {
				*savePtr -= amount;
				*checkPtr += amount;
				cout << "\nTransaction completed\n";
				sleep(1);
		}
		return;    
}

// withdraw menu option
void withdraw(float* checkPtr, float* savePtr) {
		int option;

		cout << "\nAccounts\n";
		cout << "1) Checking account\n";
		cout << "2) Savings account\n";
		cout << "[Enter any other number to return to main menu]\n\n";
		cout << "Which account would you like to withraw from? ";
		cin >> option;

		switch (option) {
				case 1:
						// withdraw from checking
						withdrawMoney(checkPtr);
						break;
				case 2:
						// withdraw from saving
						withdrawMoney(savePtr);
						break;
				default:
						return;
		}
		return;
}

// function to withdraw money
void withdrawMoney(float* fPtr) {
		float amount;

		cout << "\nHow much would you like to withdraw?\n";
		cout << "[To return to the main menu enter zero]\n";
		cout << "Amount $ ";
		cin >> amount;

		if (amount == 0) {
				return;
		}
		if (amount > *fPtr || amount < 0) {
				cout << "\nTransaction not completed.\n";
				sleep(1);
				return;
		} else {
				*fPtr -= amount;
				cout << "\n$" << setprecision(2) << fixed << amount << " has been withdrawn\n";
				cout << "You have $" << setprecision(2) << fixed << *fPtr << " remaining in this account.\n";
				sleep(3);
		}
		return;   
}